scarb/tests/build_cairo_plugin.rs

```diff
@@ -1536,3 +1536,182 @@ fn can_expand_impl_inner_func_attrr() {
+        "#})
+        .build(&project);
+
+    Scarb::quick_snapbox()
+        .arg("cairo-run")
+        // Disable output from Cargo.
+        .env("CARGO_TERM_QUIET", "true")
+        .current_dir(&project)
+        .assert()
+        .success();
+}
+
+#[test]
+fn can_use_quote_with_token_stream() {
+    let temp = TempDir::new().unwrap();
+    let t = temp.child("some");
+    CairoPluginProjectBuilder::default()
+      .add_dep(r#"cairo-lang-quote = { path="/Users/mateusz/SWM/Starkware/scarb/plugins/cairo-lang-quote" }"#)
+      .add_dep(r#"cairo-lang-primitive-token = "1.0.0""#)
+      .lib_rs(indoc! {r##"
+        use cairo_lang_macro::{ProcMacroResult, TokenStream, inline_macro, TokenTree, Token};
+        use cairo_lang_quote::quote;
+
+        #[inline_macro]
+        pub fn some(_token_stream: TokenStream) -> ProcMacroResult {
+          let token = TokenStream::new(vec![TokenTree::Ident(Token::new("5".to_string(), None))]);
+          let tokens = quote! {
+            #token
+          };
+          ProcMacroResult::new(tokens)
+        }
+      "##})
+      .build(&t);
+    let project = temp.child("hello");
+    ProjectBuilder::start()
+        .name("hello")
+        .version("1.0.0")
+        .dep("some", &t)
+        .lib_cairo(indoc! {r#"
+            fn main() -> felt252 { 
+              some!()
+            }
+        "#})
+        .build(&project);
+
+    Scarb::quick_snapbox()
+        .arg("cairo-run")
+        // Disable output from Cargo.
+        .env("CARGO_TERM_QUIET", "true")
+        .current_dir(&project)
+        .assert()
+        .success();
+}
+
+#[test]
+fn can_use_quote_with_syntax_node() {
+    let temp = TempDir::new().unwrap();
+    let t = temp.child("some");
+    CairoPluginProjectBuilder::default()
+      .add_dep(r#"cairo-lang-quote = { path="/Users/mateusz/SWM/Starkware/scarb/plugins/cairo-lang-quote" }"#)
+      .add_dep(r#"cairo-lang-primitive-token = "1.0.0""#)
+      .add_dep(r#"cairo-lang-syntax = { git = "https://github.com/starkware-libs/cairo", rev = "f3c98f7e500fdd320d531187fd02842a0114b99e" }"#)
+      .add_dep(r#"cairo-lang-parser = { git = "https://github.com/starkware-libs/cairo", rev = "f3c98f7e500fdd320d531187fd02842a0114b99e" }"#)
+      .lib_rs(indoc! {r##"
+        use cairo_lang_macro::{ProcMacroResult, TokenStream, attribute_macro};
+        use cairo_lang_quote::quote;
+        use cairo_lang_parser::utils::SimpleParserDatabase;
+        use cairo_lang_syntax::node::with_db::SyntaxNodeWithDb;
+
+        #[attribute_macro]
+        pub fn some(_attr: TokenStream, token_stream: TokenStream) -> ProcMacroResult {
+          let db_val = SimpleParserDatabase::default();
+          let db = &db_val;
+          let code = r#"
+              fn main() -> felt252 {
+                5
+              }
+          "#;
+          let syntax_node = db.parse_virtual(code).unwrap();
+          let syntax_node_with_db = SyntaxNodeWithDb::new(&syntax_node, db);
+          let tokens = quote! {
```

Comment:
I'm missing a test that is using/quoting Cairo-specific syntax, like `impl of`

Reply:
Added tests for `impl of`, coming from either syntax_node, or putting it directly into the quote